core/include/drivers/tpm2/tpm2_cmd.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Linaro Limited
+ */
+
+#ifndef __DRIVERS_TPM2_CMD_H
+#define __DRIVERS_TPM2_CMD_H
+
+#include <io.h>
+#include <stdint.h>
+
+struct tpm2_cmd_hdr {
```

Comment:
These structs looks a bit scary. It seems that they are here just for the inline functions below and apart from that only internal us in `tpm2_cmd.c`. A comment mentioning how this these structs are supposed to be used would be welcome.

Reply:
This is the command header for TPM related commands. I will add comments to describe it more
It's mostly used in tpm2_cmd.c where tpm commands are created. It's also used in tpm2_ptp_fifo.c when tpm needs to send command or receive response.